core/services/workflows/secrets/handler.go

```diff
@@ -0,0 +1,283 @@
+package syncer
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/mitchellh/mapstructure"
+
+	"github.com/smartcontractkit/chainlink-common/pkg/types"
+	"github.com/smartcontractkit/chainlink-common/pkg/types/query"
+	"github.com/smartcontractkit/chainlink-common/pkg/types/query/primitives"
+	"github.com/smartcontractkit/chainlink-common/pkg/values"
+	"github.com/smartcontractkit/chainlink/v2/core/gethwrappers/workflow/generated/workflow_registry_wrapper"
+	"github.com/smartcontractkit/chainlink/v2/core/logger"
+)
+
+type Event interface {
+	GetSecretsURL() string
+}
+
+type event struct {
+	workflow_registry_wrapper.WorkflowRegistryWorkflowForceUpdateSecretsRequestedV1
+}
+
+func newEvent(
+	e workflow_registry_wrapper.WorkflowRegistryWorkflowForceUpdateSecretsRequestedV1,
+) Event {
+	return event{WorkflowRegistryWorkflowForceUpdateSecretsRequestedV1: e}
+}
+
+func (e event) GetSecretsURL() string {
+	return e.SecretsURL.Hex()
+}
+
+type Handler interface {
+	ForceUpdateSecrets(context.Context, Event) error
+}
+
+type handler struct {
+	orm     ORM
+	fetcher FetcherFunc
+}
+
+func newForceUpdateSecretsHandler(orm ORM, fetcher FetcherFunc) Handler {
+	return &handler{orm: orm, fetcher: fetcher}
+}
+
+func (h *handler) ForceUpdateSecrets(
+	ctx context.Context,
+	event Event,
+) error {
+	// Fetch the secrets url from ORM
+	url, err := h.orm.GetSecretsURL(ctx, event.GetSecretsURL())
```

Comment:
Is this correct ? You're fetching the secrets URL from the event, and then are getting it from the DB?

Reply:
The event only has the hash, so get the plain url from the table, then fetch the contents, then write